server/build.gradle

```diff
@@ -75,8 +80,32 @@ dependencies {
     testImplementation project(':client')
 }
 
-task createVersionFile(type:Exec) {
-    commandLine 'python', 'src/main/java/com/defold/extender/version.py'
+task createVersionFile() {
+    def sha = System.getenv("COMMIT_SHORT_SHA")
+    if (sha == null) {
+        def outputStream = new ByteArrayOutputStream()
+        exec {
+            commandLine "git", "rev-parse", "--short", "HEAD"
+            standardOutput = outputStream
+        }
+        sha = outputStream.toString().trim()
+    }
+    def build_time = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date ())
+    def text = """
+// Auto generated by gradle
+
+package com.defold.extender;
+public class Version {
+    public static final String gitVersion = "${sha}";
+    public static final String buildTime = "${build_time}";
+}
+"""
+    def binding = ["sha" : sha, "build_time" : build_time]
+
+    def engine = new groovy.text.SimpleTemplateEngine()
+    def template = engine.createTemplate(text).make(binding)
+
+    new File(projectDir.toString() + "/src/main/java/com/defold/extender", "Version.java").text = template.toString()
```

Comment:
What was the benefit of this as opposed to the python script?

Reply:
I was debating wether to change this or not. Removing the python script drops a dependency on python in favour of Java.